VideoLocker/src/test/java/org/edx/mobile/view/CourseUnitNavigationActivityTest.java

```diff
@@ -502,15 +511,17 @@ public void testUnitFragmentCreation() {
             when(othersModel.getType()).thenReturn(BlockType.OTHERS);
             unitList.add(othersModel);
 
-            return Arrays.asList(new Object[][] {
-                    {unitList.get(0), CourseUnitVideoFragment.class},
-                    {unitList.get(1), CourseUnitOnlyOnYoutubeFragment.class},
-                    {unitList.get(2), CourseUnitDiscussionFragment.class},
-                    {unitList.get(3), CourseUnitMobileNotSupportedFragment.class},
-                    {unitList.get(4), CourseUnitWebViewFragment.class},
-                    {unitList.get(5), CourseUnitEmptyFragment.class},
-                    {unitList.get(6), CourseUnitMobileNotSupportedFragment.class},
-                    {unitList.get(7), CourseUnitMobileNotSupportedFragment.class}
+            return Arrays.asList(new Object[][]{
+                    {unitList.get(0), CourseUnitVideoFragment.class, true},
+                    {unitList.get(1), CourseUnitOnlyOnYoutubeFragment.class, true},
+                    {unitList.get(2), CourseUnitDiscussionFragment.class, true},
+                    // Same object, different config
+                    {unitList.get(2), CourseUnitMobileNotSupportedFragment.class, false},
+                    {unitList.get(3), CourseUnitMobileNotSupportedFragment.class, true},
+                    {unitList.get(4), CourseUnitWebViewFragment.class, true},
+                    {unitList.get(5), CourseUnitEmptyFragment.class, true},
+                    {unitList.get(6), CourseUnitMobileNotSupportedFragment.class, true},
+                    {unitList.get(7), CourseUnitMobileNotSupportedFragment.class, true}
```

Comment:
Again, instead of using indices to join the provided and expected properties in a hardcoded manner, it would be much more robust to construct the params list as you're defining the models.

```` java
List<Object[]> argsList = new ArrayList<>();
{
    VideoBlockModel encodeVideosModel = Mockito.mock(VideoBlockModel.class);
    VideoData videoData = Mockito.mock(VideoData.class);
    videoData.encodedVideos = Mockito.mock(EncodedVideos.class);
    when(videoData.encodedVideos.getPreferredVideoInfo())
            .thenReturn(Mockito.mock(VideoInfo.class));
    when(encodeVideosModel.getData()).thenReturn(videoData);
    argsList.add(new Object[] { encodeVideosModel, CourseUnitVideoFragment.class });
}
// … and so on for each argument set
return argsList;
````